proto/server.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package proto;
+
+// NotaryServer defines the TUF operations that the server supports
+service Notary {
+	rpc Auth(Token) returns (AuthResponse) {}
+	rpc Update(Updates) returns (UpdateResponse) {}
+	rpc Get(RepoRole) returns (Repository) {}
+	rpc GetKey(RepoRole) returns (KeyResponse) {}
+}
+
+// Token holds a JWT that the Notary server will check for permissions
+message Token {
+	bytes jwt = 1;
+}
+
+message Updates {
+	bytes jwt = 1;
+	string gun = 2;
+	Repository repo = 3;
+}
+
+message RepoRole {
+	bytes jwt = 1;
+	string gun = 2;
+	string role = 3;
+}
+
+// Repository is a collection of TUF files
```

Comment:
There is only one difference, and I have discussed it with the TUF designers prior to implementation. They said they would work it in to a future spec update.

In the official spec, the timestamp and snapshot files reference the other role files with a file extension, possibly including a compression extension, such as "snapshot.json.zip", with multiple entries for a role, each with different compression. The need to translate back and forth from the role name "snapshot" to the filename, and to do things like prefix matching and extension parsing to determine the right thing to request, creates a terrible chunk of code.

Instead, our timestamp and snapshot files simply contain the bare role name and compression is done at a lower level. We discussed if we could see any security vulnerabilities with that approach and couldn't come up with anything (there are discussed potential exploits in compression libraries but we didn't determine there was anything worrying to us for this use case, and major applications , like Chrome, are happy to assume decompression is a safe operation).